pkg/bitmap/bitmap.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bitmap provides the implementation of bitmap.
+package bitmap
+
+import (
+	"math"
+	"math/bits"
+)
+
+// Bitmap struct contains numOnes which represent number of ones in the
+// bitmap and a slice bitBlock which represent bits. The type of bitBlock
+// is uint64 which means each number in bitBlock can contain information
+// for 64 bits.
+//
+// +stateify savable
+type Bitmap struct {
+	numOnes  uint32
+	bitBlock []uint64
+}
+
+// NewBitmap create a new empty Bitmap.
+func NewBitmap(size uint32) *Bitmap {
```

Comment:
Thanks, @kevinGC , I have some confusion here. 
> To avoid unnecessary allocation

Does this mean that avoid allocating data on the heap? 

> `func WithSize(size uint32) Bitmap {`

Is the function named with `WithSize` siffnix, like, `BitmapWithSize` or just named as `WithSize` ?

Reply:
Yes, it avoids allocating on the heap. And since there's no specific reason to use `*Bitmap`, `Bitmap` is simpler and faster.

And either `WithSize` or `BitmapWithSize` is fine. I was trying to avoid the "stutter" in `bitmap.BitmapWithSize`, but it's up to preference and I don't have a strong opinion.